lib/Doctrine/DBAL/DBALException.php

```diff
@@ -28,21 +22,28 @@ class DBALException extends Exception
      */
     public static function driverExceptionDuringQuery(Driver $driver, Throwable $driverEx, $sql, array $params = [])
     {
-        $msg = "An exception occurred while executing '" . $sql . "'";
-        if ($params) {
-            $msg .= ' with params ' . self::formatParameters($params);
-        }
-        $msg .= ":\n\n" . $driverEx->getMessage();
+        $messageFormat = <<<'MESSAGE'
+An exception occurred while executing "%s"%s:
+
+%s
+MESSAGE;
 
-        return static::wrapException($driver, $driverEx, $msg);
+        $message = sprintf(
+            $messageFormat,
+            $sql,
+            $params !== [] ? sprintf(' with params %s', (new FormatArray())->__invoke($params)) : '',
```

Comment:
Looks like it's the only use case for `FormatArray`. Can we leave the implementation here and not introduce the class? Embedding query parameters into an exception is a bad idea in general from the security perspective, so let's not make any improvements in this area.